Replace images in wxRendererMac::DrawTitleBarBitmap() with drawing code.

TcT2k opened the pull request:
The low resolution (14x14) bitmaps scaled badly on high resolution displays. A close button suitable for usage inside a window (like `wxInfoBar`) is not available via HI theme drawing methods. This drawing code tries to emulate a close button, as close as possible to the one found in the Xcode 6+ welcome window.

Normal: <img width="77" alt="screen shot 2015-09-25 at 15 20 33" src="https://cloud.githubusercontent.com/assets/5075894/10101986/6ea7d3fc-639c-11e5-8611-fdcab76323bc.png"> Hover: <img width="69" alt="screen shot 2015-09-25 at 15 20 41" src="https://cloud.githubusercontent.com/assets/5075894/10101987/6ea8cd8e-639c-11e5-9970-37c4b6db0e12.png"> Old implementation: <img width="70" alt="screen shot 2015-09-25 at 15 43 48" src="https://cloud.githubusercontent.com/assets/5075894/10101988/6eaa52e4-639c-11e5-8778-1c3b421bd652.png">

TcT2k committed:
Replace images in wxRendererMac::DrawTitleBarBitmap() with drawing code.

The low resolution (14x14) bitmaps scaled badly on high resolution displays. A close button suitable for usage inside a window (like wxInfoBar) is not available via HI theme drawing methods. This drawing code tries to emulate a close button, as close as possible to the one found in the Xcode 6+ welcome window.

TcT2k committed:
Handle high resolution displays in wxBitmapButtonBase::NewCloseButton().

Bitmap for the close button is scaled based on GetContentScaleFactor(). On OS X the resulting bitmap has to use the correct scaling factor in order to be displayed correctly on the button when using a high resolution display.

TcT2k commented:
@vadz previous comments where lost when I pushed the updated implementation, so I'll reply here:
I wasn't even ware that `GetContentScaleFactor()` returns something on MSW. I've updated the second commit to include other ports too. It looks much better now on Windows. But a little bit of platform specific code is still required as only the OSX version of `wxBitmap` has ctors with a specifiable content scaling factor.

Windows 8.1 (200%):
![wxwidgets_closebtn_win](https://cloud.githubusercontent.com/assets/5075894/10304616/8244d64e-6c1b-11e5-8f81-f92674cf5fee.png)

vadz commented on src/osx/carbon/renderer.cpp:
```diff
@@ -699,218 +699,55 @@ void wxRendererMac::DrawTitleBarBitmap(wxWindow *win,
+    dc.SetPen(wxPen(glyphColor, 1 * contentScale));
+    
+    wxRect centerRect(rect);
+    centerRect.Inflate(-5 * contentScale);
```
This could be written a bit more clearly as `Deflate(5*contentScale)`.

vadz commented on src/osx/carbon/renderer.cpp:
```diff
@@ -699,218 +699,55 @@ void wxRendererMac::DrawTitleBarBitmap(wxWindow *win,
-        data = close_pressed_png;
-        len = WXSIZEOF(close_pressed_png);
+        drawCircle = true;
+        glyphColor = wxColour(104, 104, 104);
```
It would be nice to know where do all these hardcoded RGB values come from.

vadz commented on src/common/bmpbtncmn.cpp:
```diff
@@ -103,13 +103,21 @@ GetCloseButtonBitmap(wxWindow *win,
     wxRendererNative::Get().
-        DrawTitleBarBitmap(win, dc, size, wxTITLEBAR_BUTTON_CLOSE, flags);
+        DrawTitleBarBitmap(win, dc, drawSize, wxTITLEBAR_BUTTON_CLOSE, flags);
+#ifdef __WXOSX__
```
Wouldn't using `wxBitmap::CreateScaled()` make it possible to do this without any `#ifdefs`?

vadz commented:
Sorry for the delay, I've finally reviewed the latest version of the patch. If the latest comments could be addressed (especially the one about `CreateScaled()`), I promise to apply this without any further delay. TIA!

TcT2k commented:
I've integrated your comments, but I wasn't able to use `wxBitmap::CreateScaled()` (which is currently undocumented) as the result was an empty bitmap. I'm not really sure why as the code looks correct when stepping into it. It would obviously be cleaner without the #ifdef but I couldn't find a way to make it work without the cocoa specific ctor.

vadz commented:
It would be great if somebody with a retina Mac could debug this, it's really appalling that we don't have a way to create a bitmap in a portable way :-(

Also, sorry to be a pain, but what about the hardcoded RGB values? They seem to be there without any explanations...

TcT2k committed:
Document wxBitmap::CreateScaled() and wxBitmap::Create() overload.

The two methods were previously undocumented but might be of interest
to user code for high DPI display compatibility.

TcT2k commented:
@vadz 
I've misinterpreted `CreateScaled`. I thought it would create a scaled bitmap based and the current content, but obviously it creates the bitmap object like other `Create` methods.
The code now works without ifdefs. I've tested OS X and Win 8.1 with a 4K display and the results match the screenshots posted above.

As for the RGB values I've included a comment that they are hard coded based on the Xcode 6+ welcome screen. I haven't found a good match using any system colors, which would obviously be a nicer approach.

I've also added documentation for `CreateScaled`in an additional commit (3ab3acd).

vadz commented:
Great, thanks for fixing this!
